source/include/core_mqtt.h

```diff
@@ -26,11 +26,16 @@
 #ifndef CORE_MQTT_H
 #define CORE_MQTT_H
 
-/* Include main config file before other headers. */
-#include "core_mqtt_config.h"
-
-/* Include config defaults file after main config file to get
- * default values of configs not defined in main config file. */
+/* mqttDO_NOT_USE_CUSTOM_CONFIG allows building the MQTT library
+ * without a custom config. If a custom config is provided, the
+ * mqttDO_NOT_USE_CUSTOM_CONFIG macro should not be defined. */
+#ifndef mqttDO_NOT_USE_CUSTOM_CONFIG
```

Comment:
Why is mqtt lowercase? This is different than other macros in the library.

Reply:
Sure, changed it. Was just biased by the naming pattern here: https://github.com/aws/amazon-freertos/blob/v1.4.8/lib/include/aws_mqtt_lib.h#L38

Reply:
I'm not sure why this is necessary.  Can the user just leave the config file empty if they want to use the default values?

Reply:
We discussed about it internally, and thought that it would be good to not force the customer to define a config file if they don't want to define custom values. 
We decided to mimic the approach that was followed in a previous MQTT release: https://github.com/aws/amazon-freertos/blob/release-1.4.x/lib/include/aws_mqtt_lib.h#L38

Reply:
We discussed about it separately, and agreed that this is the best option we have of all options considered for customer usability experience. The concern of ambiguity for the customer is mitigated with clear information on the REAMDE about the options of building the library with or without config file